### PR TITLE
fix!: unable to use `test` without `bring cloud`

### LINKED ISSUE #2521: Without "bring cloud" I am getting "ENOENT: no such file or directory" when using `test` statement
### I tried this:

```
test "test" {

};
```

### This happened:

```
$ wing compile -t sim example.w
ENOENT: no such file or directory, open ''
```

### I expected this:

It compiles, or gives me an error that the statement requires `bring cloud;` to be added to the beginning of the file

### Is there a workaround?

Add `bring cloud`

### Component

Compiler

### Wing Version

0.17.3

### Wing Console Version

_No response_

### Node.js Version

18.14.1

### Platform(s)

MacOS

### Anything else?

_No response_

### Community Notes

* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.

### LINKED ISSUE #2598: Unknown symbol "cloud"
### I tried this:

Starting with this code

```
bring "cdktf" as cdktf;

let bucketModule = new cdktf.TerraformHclModule(
  source:  "terraform-aws-modules/s3-bucket/aws",
  variables: {
    bucket_prefix: "wing-it",
    // the cdktf.Token.asString() function is a workaround for
    // this issue https://github.com/winglang/wing/issues/2597
    force_destroy: cdktf.Token.asString(true),
    versioning: cdktf.Token.asString({
      enabled: true
    })
  }
);
```

I added a test

```
bring "cdktf" as cdktf;

let bucketModule = new cdktf.TerraformHclModule(
  source:  "terraform-aws-modules/s3-bucket/aws",
  variables: {
    bucket_prefix: "wing-it",
    // the cdktf.Token.asString() function is a workaround for
    // this issue https://github.com/winglang/wing/issues/2597
    force_destroy: cdktf.Token.asString(true),
    versioning: cdktf.Token.asString({
      enabled: true
    })
  }
);

test "Hello, world!" {
}
```

### This happened:

I got an error in VS Code: `Unknown symbol "cloud"`

Screenshot 2023-05-20 at 22 42 57

When compiling, a slightly different error 

```
wing compile -t tf-aws --debug  main.w
ENOENT: no such file or directory, open ''
```

### I expected this:

- In the best case no error 
- Second best would be a more helpful error message in the editor and the CLI

### Is there a workaround?

Add `bring cloud;`

### Component

Compiler

### Wing Version

0.17.16

### Wing Console Version

0.17.16

### Node.js Version

v18.16.0

### Platform(s)

MacOS

### Anything else?

_No response_

### Community Notes

* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.

### LINKED ISSUE #2619: calling `test` without `bring cloud` results in an unclear error
### I tried this:

running a test without calling `bring cloud;` at the beginning of the file:

```
test "test" {
    log("test");
}
```

### This happened:

`ENOENT: no such file or directory, open ''`
image

### I expected this:

To throw a more explicit error- telling me I forgot to bring cloud 

### Is there a workaround?

_No response_

### Component

Compiler

### Wing Version

_No response_

### Wing Console Version

_No response_

### Node.js Version

_No response_

### Platform(s)

_No response_

### Anything else?

_No response_

### Community Notes

* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.

### LINKED ISSUE #2702: std is not defined error when accessing `std.Duration`
### I tried this:

Tried to create a schedule resource using `rate`.

```
let schedule = new cloud.Schedule(cloud.ScheduleProps{
  rate: std.Duration.fromMinutes(5),
});
```

### This happened:

image

### I expected this:

The Schedule resource to be created.

### Is there a workaround?

just using Schedule with `cron` works.

### Component

Compiler

### Wing Version

_No response_

### Wing Console Version

_No response_

### Node.js Version

_No response_

### Platform(s)

_No response_

### Anything else?

_No response_

### Community Notes

* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.

### PR DESCRIPTION
**Pull request by eladb**
Move `cloud.Test` to `std.Test` based on the approach that `std` includes types that are implicitly imported ("brought?"). But... for some reason we have special-cases for *every* type in `std` when we jsifiy, so now we are implicitly converting references to `std` during jsification to `$stdlib.std`.

As a follow up we might be able to get rid of many of the special cases we have for `std` types, but I'll save that for a future refactor.

## Fixes

Fixes #2521
Fixes #2619
Fixes #2598
Fixes #2702

## Misc

- Improve diagnostics for internal compiler errors (throw an error instead of `process.exit()`!).
- Only use diagnostic error formatter if `span.file_id` is not empty. The reason we didn't have diagnostics in the case where `cloud` was not brought was because it's span was empty.

## Checklist

- [x] PR title matches [Winglang's style guide](https://docs.winglang.io/contributors/pull_requests#how-are-pull-request-titles-formatted)
- [x] PR description explains motivation and solution
- [x] Tests added
- [x] Docs updated
- [x] Needs end-to-end tests (`pr/e2e-full` label)

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Monada Contribution License](https://docs.winglang.io/terms-and-policies/contribution-license.html)*.

### CODE REVIEW

File: examples/tests/sdk_tests/bucket/events.w [hunk 1]
```diff
@@ -20,8 +20,8 @@ b.onEvent(inflight (key: str) => {
     counter.inc();
 });
 
-
-new cloud.Test(inflight () => {
+// "std" is implicitly imported
```
Comment (eladb) [inline examples/tests/sdk_tests/bucket/events.w:23]: yes, that's a bit ugly

**Comment by Chriscbr** [comment]
This might also fix #2702?

File: examples/tests/valid/test_without_bring.w [hunk 2]
```diff
@@ -0,0 +1,3 @@
+test "hello test" {
+  assert(true);
+}
```
Comment (Chriscbr) [inline examples/tests/valid/test_without_bring.w:3]: Adding new tests increases our CI build times by a bit, we could also just remove `bring cloud;` from [this file](https://github.com/winglang/wing/blob/main/examples/tests/valid/std_string.w)

File: libs/wingc/src/jsify.rs [hunk 3]
```diff
@@ -251,12 +251,19 @@ impl<'a> JSifier<'a> {
 	}
 
 	fn jsify_user_defined_type(&self, user_defined_type: &UserDefinedType) -> String {
+		// "std" is implicitly imported, so replace with "$stdlib.std".
+		let root = if user_defined_type.root == WINGSDK_STD_MODULE.into() {
+			format!("{STDLIB}.{WINGSDK_STD_MODULE}")
+		} else {
+			user_defined_type.root.to_string()
+		};
```
Comment (Chriscbr) [inline libs/wingc/src/jsify.rs:259]: Is there a way to avoid adding this edge case? I think it might be possible to remove the `$stdlib` identifier we made up, and use `std` directly

File: examples/tests/valid/test_without_bring.w (hunk 2 above)
Comment (eladb) [reply examples/tests/valid/test_without_bring.w:3]: Adding new tests should be "free". If it's not, let's run do more parallelism.

File: libs/wingc/src/jsify.rs (hunk 3 above)
Comment (eladb) [reply libs/wingc/src/jsify.rs:259]: I kind of like this solution. Let me look into it.

File: examples/tests/valid/test_without_bring.w (hunk 2 above)
Comment (eladb) [reply examples/tests/valid/test_without_bring.w:3]: At any rate. I am leaving this test because I want to make sure we have explicit coverage for this case.

**Comment by eladb** [comment]
> This might also fix #2702?

Yap
